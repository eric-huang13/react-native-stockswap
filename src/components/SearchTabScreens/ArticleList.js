import React, {Component} from 'react';
import {
  View,
  SafeAreaView,
  TextInput,
  ScrollView,
  StyleSheet,
  FlatList,
  Text,
  RefreshControl,
  TouchableOpacity,
  Linking
} from 'react-native';
import {connect} from 'react-redux';
import Article from '../SearchTabComponents/Article';
import SearchInput from '../../icons/SearchInput';
import {moderateScale} from '../../util/responsiveFont';
import {fetchNews} from '../../actions/news'


export class ArticleList extends Component {

  componentDidMount() {
    this.props.fetchNews();
  }
  constructor(props) {
    super(props);
    this.state = {
      input: '',
    };
  }
  handleChange = (text) => {
    this.setState({input: text});
  };

   //fetches more data, increases page number by 1
   fetchMoreData = () => {
    if (!this.props.loading && this.props.loadMore){
    // this.props.fetchNews(this.props.page, this.props.offset);
    this.props.fetchNews();

    console.log(this.props.page,'ran')
    }
  };
  render() {
    const {articles} = this.props;
    // console.log(articles,"NEWS")
    // const filteredArticles = articles.filter((item) =>
    //   item.headline.toLowerCase().includes(this.state.input.toLowerCase()),
    // );
    return (
      <SafeAreaView>
        {/* <ScrollView contentContainerStyle={{paddingBottom: 220}}> */}
        <FlatList
        ListHeaderComponent={
          <View style={style.searchInputContainer}>
            <View
              style={{
                position: 'absolute',
                zIndex: 1,
                left: moderateScale(14),
                top: moderateScale(10),
              }}>
              <SearchInput />
            </View>
            <TextInput
              style={style.searchInput}
              placeholder="Search"
              placeholderTextColor="lightgrey"
              onChangeText={(text) => this.handleChange(text)}
            />
          </View>
          }
          keyExtractor={(item, index) => String(index)}
          contentContainerStyle={{ paddingBottom: 20 }}
          data={this.props.articles}
           renderItem={({item, index}) => ( 
            <TouchableOpacity
                key={item.id}
                onPress={() => {
                  Linking.openURL(item.news_url);
                }}>
            <Article key={item.id} item={item} />
              </TouchableOpacity>
           )}  
           refreshControl={
            <RefreshControl
              //refresh control used for the Pull to Refresh
              refreshing={this.props.loading}
              onRefresh={this.fetchMoreData}
            />
          }
          //Will uncomment if pagination is set up
          //  onEndReachedThreshold={0.1} 
          //  onEndReached={this.fetchMoreData}
        ListFooterComponent={
          <View style={style.footer}>
            {this.props.loading ? <Text style={style.loadingText}>LOADING...</Text> : null}
          </View>
        }                   
          />

          {/* {filteredArticles.map((item) => {
            return <Article key={item.id} item={item} />;
          })} */}
        {/* </ScrollView> */}
      </SafeAreaView>
    );
  }
}
const mapDispatchToProps = (dispatch) => {
  return {
    fetchNews: (page, offset) => dispatch(fetchNews(page, offset)),
  };
};

const mapStateToProps = (state) => {
  return {
    articles: state.news.articles,
    loading:state.news.loading,
    page:state.news.page,
    loadMore:state.news.loadMore,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(ArticleList);

const style = StyleSheet.create({
  searchInputContainer: {
    marginBottom: moderateScale(20),
  },
  searchInput: {
    paddingLeft: moderateScale(40),
    alignContent: 'center',
    backgroundColor: '#3e4d6c',
    color: 'lightgrey',
    fontSize: moderateScale(16),
    height: moderateScale(36),
    paddingVertical: 0,
    fontFamily: 'Montserrat-Italic',
  },
  loadingText:{
    color:'#8B64FF',
    textAlign:"center",
    fontSize:18,
  },
  footer:{
    height: 0, 
    marginBottom: 210,
  }
});
