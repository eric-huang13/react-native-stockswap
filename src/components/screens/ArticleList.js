import React, {Component} from 'react';
import {View, TextInput, ScrollView} from 'react-native';
import {connect} from 'react-redux';
import Article from './Article';

export class ArticleList extends Component {
  constructor(props) {
    super(props);
    this.state = {
      input: '',
    };
  }
  handleChange = (text) => {
    this.setState({input: text});
  };
  render() {
    const {articles} = this.props;
    const filteredArticles = articles.filter((item) =>
      item.headline.toLowerCase().includes(this.state.input.toLowerCase()),
    );
    return (
      <View>
        <ScrollView contentContainerStyle={{paddingBottom:180}}>
        <View>
          <TextInput
            style={{borderWidth: 0.5, marginHorizontal: 1}}
            placeholder="Search by name"
            onChangeText={(text) => this.handleChange(text)}
          />
        </View>
        {filteredArticles.map((item) => {
          return <Article key={item.id} item={item} />;
        })}
        </ScrollView>
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    articles: state.company.articles,
  };
};

export default connect(mapStateToProps)(ArticleList);
