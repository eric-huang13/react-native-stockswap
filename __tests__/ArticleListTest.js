import React from 'react';
import ArticleList from '../src/components/SearchTabScreens/ArticleList';
import renderer from 'react-test-renderer';


jest.mock("@react-native-async-storage/async-storage", () =>
  require("@react-native-async-storage/async-storage/jest/async-storage-mock"),
);
jest.mock('react-redux', () => ({
    connect: () => {
      return (component) => {
        return component
      };
    }
  }));

test('renders correctly', () => {
    const tree = renderer.create(<ArticleList/>).toJSON();
  expect(tree).toMatchSnapshot();
});