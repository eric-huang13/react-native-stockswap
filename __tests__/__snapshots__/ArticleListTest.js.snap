// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <RCTScrollView
    ListFooterComponent={
      <View
        style={
          Object {
            "height": 0,
            "marginBottom": 210,
          }
        }
      />
    }
    ListHeaderComponent={
      <View
        style={
          Object {
            "marginBottom": 28.248175182481752,
          }
        }
      >
        <View
          style={
            Object {
              "left": 19.77372262773723,
              "position": "absolute",
              "top": 14.124087591240876,
              "zIndex": 1,
            }
          }
        >
          <SearchInput />
        </View>
        <Component
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Search"
          placeholderTextColor="lightgrey"
          rejectResponderTermination={true}
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#3e4d6c",
              "color": "lightgrey",
              "fontFamily": "Montserrat-Italic",
              "fontSize": 22.598540145985403,
              "height": 50.846715328467155,
              "paddingLeft": 56.496350364963504,
              "paddingVertical": 0,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
    }
    contentContainerStyle={
      Object {
        "paddingBottom": 20,
      }
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.1}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "marginBottom": 28.248175182481752,
            }
          }
        >
          <View
            style={
              Object {
                "left": 19.77372262773723,
                "position": "absolute",
                "top": 14.124087591240876,
                "zIndex": 1,
              }
            }
          >
            <RNSVGSvgView
              align="xMidYMid"
              bbHeight={25.423357664233578}
              bbWidth={25.423357664233578}
              focusable={false}
              height={25.423357664233578}
              meetOrSlice={0}
              minX={0}
              minY={0}
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  Object {
                    "flex": 0,
                    "height": 25,
                    "width": 25,
                  },
                ]
              }
              vbHeight={18}
              vbWidth={18}
              width={25.423357664233578}
              xmlns="http://www.w3.org/2000/svg"
            >
              <RNSVGGroup>
                <RNSVGDefs />
                <RNSVGGroup
                  fill={null}
                  fillRule={0}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      -346,
                      -529,
                    ]
                  }
                  propList={
                    Array [
                      "fill",
                      "fillRule",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M362.805 544.862l-5.114-5.109a5.97 5.97 0 001.32-3.753c0-3.314-2.689-6-6.005-6a6.003 6.003 0 00-6.006 6c0 3.314 2.689 6 6.006 6a5.983 5.983 0 003.74-1.306l5.115 5.11a.668.668 0 00.944-.942zM348.335 536a4.67 4.67 0 014.67-4.667 4.67 4.67 0 014.672 4.667 4.65 4.65 0 01-1.376 3.307s0 0 0 0 0 0 0 0a4.659 4.659 0 01-3.295 1.36 4.67 4.67 0 01-4.671-4.667z"
                    fill={4288192172}
                    fillRule={1}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                        "stroke",
                      ]
                    }
                    stroke={4288192172}
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Search"
            placeholderTextColor="lightgrey"
            rejectResponderTermination={true}
            style={
              Object {
                "alignContent": "center",
                "backgroundColor": "#3e4d6c",
                "color": "lightgrey",
                "fontFamily": "Montserrat-Italic",
                "fontSize": 22.598540145985403,
                "height": 50.846715328467155,
                "paddingLeft": 56.496350364963504,
                "paddingVertical": 0,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "height": 0,
              "marginBottom": 210,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
